src/Symfony/Bundle/SecurityBundle/Command/UserPasswordEncoderCommand.php

```diff
@@ -85,14 +88,16 @@ protected function execute(InputInterface $input, OutputInterface $output)
             $password = $helper->ask($input, $output, $passwordQuestion);
         }
 
-        if (!$salt) {
+        if (!$salt && !$input->getOption('no-interaction')) {
```

Comment:
I think you can use [isInteractive](https://github.com/symfony/symfony/blob/2.7/src/Symfony/Component/Console/Input/Input.php#L86) here ?

```php
if (!$salt) {
    if ($input->isInteractive()) {
        $saltQuestion = $this->createSaltQuestion($input, $output);
        $salt = $helper->ask($input, $output, $saltQuestion);
    } else {
        $salt = $this->generateSalt($output);
    }
}
```